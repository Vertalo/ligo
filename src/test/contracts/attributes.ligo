const x: int = 1; attributes ["inline"];

function foo (const a : int) : int is
  block {
    const test: int = 2 + a; attributes ["inline"];
  } with test;
attributes ["inline"];
  
const y: int = 1; attributes ["inline"; "other"];

function bar (const b : int) : int is
  block {
    function test (const z : int) : int is begin
      const r : int = 2 + b + z
    end with r;
    attributes ["inline"; "foo"; "bar"];
  } with test(b);
