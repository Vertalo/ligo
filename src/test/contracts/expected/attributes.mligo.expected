let x = 1 [@@inline]

let foo (a : int) : int =
  (let test = 2 + a [@@inline] in
   test) [@@inline]

let y = 1 [@@inline][@@other]

let bar (b : int) : int =
  let test = fun (z : int) -> 2 + b + z
    [@@inline]
    [@@foo]
    [@@bar] in
  test b
